source/common/http/http1/codec_impl.cc

```diff
@@ -552,7 +588,12 @@ void ConnectionImpl::onMessageCompleteBase() {
     http_parser_pause(&parser_, 1);
     return;
   }
-  onMessageComplete();
+
+  if (header_parsing_state_ == HeaderParsingState::Value) {
```

Comment:
Can you add a small comment here that this happens when there were trailers?

Reply:
Thanks for the comment! 
If you have any insight into why this happens here, where in the headers case it gets called elsewhere it might be helpful.  I think Matt had enough context to know offhand but as I don't I'm going to assume other folks touching the codec might not either :-)

Reply:
Sorry missed your comment here @alyssawilk. I agree, I had to ask the folks over at `http_parser` for details on how the callback works. I'll add something to the header of this file?